main.js

```diff
@@ -0,0 +1,802 @@
+              }
+            });
+            break;
+
+          case 'ArrowUp':
+            keyElement.classList.add('dark');
+            keyElement.innerHTML = '<i class="fas fa-arrow-up"></i>';
+            keyElement.addEventListener('click', () => {
+              if (this.properties.shift) {
+                this.properties.shift = false;
+                this.toggleShift();
+              }
+              const lineBreakIdx = textarea.value.indexOf('\n', -this.getCaretPosition());
+              const caretIndex = lineBreakIdx - Math.abs(lineBreakIdx - this.getCaretPosition());
+              this.setCaretPosition(caretIndex);
+            });
+            break;
+
+          case 'ArrowLeft':
+            keyElement.classList.add('dark');
+            keyElement.innerHTML = '<i class="fas fa-arrow-left"></i>';
+            keyElement.addEventListener('click', () => {
+              if (this.properties.shift) {
+                this.properties.shift = false;
+                this.toggleShift();
+              }
+              this.setCaretPosition(this.getCaretPosition() - 1);
+            });
+            break;
+
+          case 'ArrowDown':
+            keyElement.classList.add('dark');
+            keyElement.innerHTML = '<i class="fas fa-arrow-down"></i>';
+            keyElement.addEventListener('click', () => {
+              if (this.properties.shift) {
+                this.properties.shift = false;
+                this.toggleShift();
+              }
+              const lineBreakIdx = textarea.value.indexOf('\n', this.getCaretPosition());
+              const caretIndex = lineBreakIdx + Math.abs(lineBreakIdx - this.getCaretPosition());
+              this.setCaretPosition(caretIndex);
+            });
+            break;
+
+          case 'ArrowRight':
+            keyElement.classList.add('dark');
+            keyElement.innerHTML = '<i class="fas fa-arrow-right"></i>';
+            keyElement.addEventListener('click', () => {
+              textarea.focus();
+              if (this.properties.shift) {
+                this.properties.shift = false;
+                this.toggleShift();
+              }
+              this.setCaretPosition(this.getCaretPosition() + 1);
+            });
+            break;
+
+          case 'CapsLock':
+            keyElement.classList.add('double', 'dark');
+            keyElement.innerHTML = '<span>CapsLock</span>';
+            keyElement.addEventListener('click', () => {
+              textarea.focus();
+              this.toggleCapsLock();
+              if (this.properties.shift) {
+                this.properties.shift = false;
+                this.toggleShift();
+              }
+            });
+            break;
+
+          default:
+            const keyValue = this.properties.lang === 'rus' ? key.value2.toLowerCase() : key.value1.toLowerCase();
+            keyElement.textContent = keyValue;
+            keyElement.addEventListener('click', () => {
```

Comment:
This is not the best way to add listeners to buttons click. You have to use event delegation for this purpose.